### PR TITLE
#18554 v1.10 backport 

### PR DESCRIPTION
**Pull request by jibi**
* https://github.com/cilium/cilium/pull/18554 -- Enable CI for feature branches (@jibi)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 18554; do contrib/backporting/set-labels.py $pr done 1.10; done
```

### CODE REVIEW

**Comment by jibi** [comment]
/test-backport-1.10

File: .github/workflows/lint-codeql.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   pull_request:
     branches:
       - v1.10
+      - ft/v1.10/**
   push:
     branches:
       - v1.10
```
Comment (kkourt) [inline .github/workflows/lint-codeql.yaml:10]: @jibi :wave: I'm going over the v1.11 backport of the same patch, and was wondering whether we need to add a line under branches here as well?

i.e., 
```
    branches:
        - v1.10
        - ft/v1.10/**
```

in the case of 1.10.

Comment (jibi) [reply .github/workflows/lint-codeql.yaml:10]: Yes, for v1.11 we'll need to change that to `ft/v1.11/**`

File: Documentation/contributing/development/contributing_guide.rst [hunk 2]
```diff
@@ -15,6 +15,10 @@ Clone and Provision Environment
 
 #. Make sure you have a `GitHub account <https://github.com/join>`_
 #. Clone the cilium repository into your ``GOPATH``.
+#. Fork the Cilium repository to your GitHub user or organization.
+#. Turn off GitHub actions for your fork as described in the `GitHub Docs <https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository>`_.
+   #. This is recommended to avoid unnecessary CI notification failures on the fork.
+#. Clone your ``${YOUR_GITHUB_USERNAME_OR_ORG}/cilium`` fork into your ``GOPATH``, and setup the base repository as ``upstream`` remote:
```
Comment (nbusseneau) [inline Documentation/contributing/development/contributing_guide.rst]: I'm not sure we should update the documentation for the backports 🤔

Comment (jibi) [reply Documentation/contributing/development/contributing_guide.rst]: Yeah, on a second thought I should probably not. Let me get rid of it

**Comment by jibi** [comment]
/test-backport-1.10

**Review by nbusseneau — APPROVED** [review]

**Comment by jibi** [comment]
Couple of unrelated flakes, marking as ready to merge

**Pull request merged**